test/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+set(SRCROOT "${PROJECT_SOURCE_DIR}/test/src")
+
+SET(SRC
+    "${SRCROOT}/Main.cpp"
+    "${SRCROOT}/UnitTests.hpp"
+    "${SRCROOT}/UnitTests.cpp"
+    "${SRCROOT}/Vector2.cpp"
+    "${SRCROOT}/Rect.cpp"
+)
+
+include_directories("${PROJECT_SOURCE_DIR}/include")
+include_directories("${PROJECT_SOURCE_DIR}/extlibs/headers")
+
+# Using sfmltest instead of test because the latter is reserved.
+add_executable(sfmltest ${SRC})
+
+target_link_libraries(sfmltest sfml-graphics sfml-window sfml-audio sfml-network sfml-system)
```

Comment:
Nothing major, but not all sfml libs need to be listed here as dependencies will automatically be linked. Could probably just list `sfml-graphics sfml-audio sfml-network`. Depends if you prefer comcise or explicit

Reply:
We always prefer explicit. Would be strange to test sfml-system and not see it in the list of linked libraries.

But this raises a more important point in my opinion: why don't we have one test executable per module? SFML modules are more or less independant, and we can select which ones are built, so we should have the same modularity for tests.

Reply:
This makes a lot of sense, especially as there are cmake options to enable/disable certain modules

Reply:
Agreed. And if you disable some modules from building, you get a linking errors

Reply:
Source files for a specific module are now only added when the module is explicitly selected to be built and only then the lib is set. Required to split the test utilities into module-based parts.